docs/usecase.rst

```diff
@@ -37,14 +72,37 @@ compute/data/communications abstractions,
 quality of service guarantees,
 and performance optimizations.
 
+.. seealso:: :doc:`executor`
+
 Generic sequential task execution
 ---------------------------------
 
 Units of work (nodes) describe the inputs and expected outputs,
 and identify a module able to perform the operation.
 
+| 1. Receive a node of work to execute.
+| 2. Import a module implementing the operation.
+| 3. Initialize or restore checkpoint state.
+| 4. Prepare operation resources.
+|    3.1 Request or confirm locality of input data.
+|    3.2 Create data publishing resource for operation.
```

Comment:
I am not sure I understand well 3.2. Could you elaborate? It is probably me missing something here, not necessarily a problem with the text.

Reply:
This _is_ a non-trivial detail. Operation implementations have different ways of producing output and require different amounts of adapter code.
We want
* a well-defined scope in which an operation can produce output, with a clear point by which results have been produced or an error has occurred
* an abstraction for the run time facet that handles output that is sufficiently flexible to be adapted to different operation implementation styles
* data events that are clearly marked and uniquely identifiable
* strong unique mapping of the run time provider and consumer of a result so that resource ownership is unambiguous

The approach we have taken is to construct a Publisher that is provided to a task when it is launched and destroyed when the task finishes. The Publisher marks the publication of the expected data events for the operation. Access to the Publisher is composed into the Resources object that is passed to the callable by which the task is executed. Depending on the Operation implementation
* the callable may pass the resources directly to a function that obtains an output object that the function can write to or call setters on,
* the callable may extract input parameters from the resources and capture the return value of another function to apply to the output part of the resource
* the callable may already have a result from another implementation detail, or may know the result in advance (such as the path to an output file that must not be considered valid until the task is complete), in which case the callable does not receive the result directly from the task it executes, but publishes the result once the task is confirmed to have executed successfully.

Reply:
Thanks. This partially informed our answer to #23. I think data event should be decomposed in: what constitute a datum in this context (e.g., file, in-memory/serialized object, etc.), how the production/availability of this datum is communicated (e.g., RPC call?), how the datum is transferred across components (e.g., different machines run different components, same machine runs all the components, etc.).